libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -1696,37 +1696,46 @@ bool RCOutput::set_neopixel_num_LEDs(const uint16_t chan, uint8_t num_leds)
 
 /*
   setup neopixel (WS2812B) output data for a given output channel
-  and mask of LEDs on the channel
+  and a LED number. LED -1 is all LEDs
 */
-void RCOutput::set_neopixel_rgb_data(const uint16_t chan, uint32_t ledmask, uint8_t red, uint8_t green, uint8_t blue)
+void RCOutput::_set_neopixel_rgb_data(pwm_group *grp, uint8_t idx, uint8_t led, uint8_t red, uint8_t green, uint8_t blue)
+{
+    const uint8_t pad_start_bits = 1;
+    const uint8_t neopixel_bit_length = 24;
+    const uint8_t stride = 4;
+    uint32_t *buf = grp->dma_buffer + (led * neopixel_bit_length + pad_start_bits) * stride + idx;
```

Comment:
Don't you need to sanitize the led number before this somehow? IE what prevents something from calling this with a value of 127?

Reply:
it is checked in caller in set_neopixel_rgb_data()